hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -371,43 +390,51 @@ SqlResult execute(DropTypePlan plan) {
         return UpdateSqlResultImpl.createUpdateCountResult(0);
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
     SqlResult execute(ShowStatementPlan plan) {
-        Stream<String> rows;
+        Stream<List<?>> rows;
 
         switch (plan.getShowTarget()) {
             case MAPPINGS:
-                rows = catalog.getMappingNames().stream();
+                rows = catalog.getMappingNames().stream().map(Collections::singletonList);
                 break;
             case VIEWS:
-                rows = catalog.getViewNames().stream();
+                rows = catalog.getViewNames().stream().map(Collections::singletonList);
                 break;
             case JOBS:
                 JetServiceBackend jetServiceBackend = nodeEngine.getService(JetServiceBackend.SERVICE_NAME);
-                rows = jetServiceBackend.getJobRepository().getActiveJobNames().stream();
+                rows = jetServiceBackend.getJobRepository().getActiveJobNames().stream().map(Collections::singletonList);
                 break;
             case TYPES:
-                rows = catalog.getTypeNames().stream();
+                rows = catalog.getTypeNames().stream().map(Collections::singletonList);
                 break;
             case DATACONNECTIONS:
                 InternalDataConnectionService service = nodeEngine.getDataConnectionService();
                 DataConnectionServiceImpl dataConnectionService = (DataConnectionServiceImpl) service;
                 rows = DataConnectionResolver
-                        .getAllDataConnectionEntries(dataConnectionService, dataConnectionCatalog.getDataConnectionStorage())
-                        .stream()
-                        .map(DataConnectionCatalogEntry::name);
+                        .getAllDataConnectionNameWithTypes(dataConnectionService)
```

Comment:
You changed `SHOW DATA CONNECTIONS` behavior, is it intended?

Reply:
Yes. Correct me if I'm wrong, but semantics will be the same - all the same connections will be shown. The only difference is that for `SHOW DATA CONNECTIONS` we add additional column to provide user an information about supported object types. Idea is from this message: https://hazelcast.slack.com/archives/CJZ745RRT/p1681908716511929?thread_ts=1681849114.133969&cid=CJZ745RRT

Reply:
this causes https://github.com/hazelcast/hazelcast/issues/24415, OBJECT TYPE without CONNECTION TYPE does not make much sense.
In slack thread we dissussed adding it to `SHOW RESOURCES` not `SHOW DATA CONNECTIONS`